src/CBot/stdlib/MathFunctions.cpp

```diff
@@ -249,6 +249,52 @@ bool rIsNAN(CBotVar* var, CBotVar* result, int& exception, void* user)
     return true;
 }
 
+// Instruction "log(number)"
+
+bool rLog(CBotVar* var, CBotVar* result, int& exception, void* user)
+{
+    switch (var->GetType())
+    {
+        case CBotTypDouble:
+            *result = log(var->GetValDouble());
+            break;
+        case CBotTypFloat:
+            *result = log(var->GetValFloat());
+            break;
+    }
+    
+    return true;
+}
+
+// Instruction "log10(number)".
+
+bool rLog10(CBotVar* var, CBotVar* result, int& exception, void* user)
+{
+    switch (var->GetType())
+    {
+        case CBotTypDouble:
+            *result = log10(var->GetValDouble());
+            break;
+        case CBotTypFloat:
+            *result = log10(var->GetValFloat());
+            break;
+    }
+    
+    return true;
+}
+
+CBotTypResult cLog(CBotVar* &var, void* user)
+{
+    if ( var == nullptr ) return CBotTypResult(CBotErrLowParam);
+    if ( var->GetType() != CBotTypFloat || var->GetType() != CBotTypDouble ) return CBotTypResult(CBotErrBadNum);
```

Comment:
This line is wrong. The parameter has to somehow simultaneously be float and double to pass this typecheck.